pkg/url/kubernetes.go

```diff
@@ -57,9 +57,13 @@ func (k kubernetesClient) ListFromCluster() (URLList, error) {
 func (k kubernetesClient) List() (URLList, error) {
 	// get the URLs present on the cluster
 	clusterURLMap := make(map[string]URL)
-	clusterURLs, err := k.ListFromCluster()
-	if err != nil {
-		return URLList{}, errors.Wrap(err, "unable to list routes")
+	var clusterURLs URLList
+	var err error
+	if k.client.GetKubeClient() != nil {
```

Comment:
Can't we check against the whole client instead of the kubeclient inside it? Also I think, if we can't check against the cluster while listing, the state should be `unknown`.

Reply:
no actually we cannot because if you observe here https://github.com/openshift/odo/pull/4472/files#diff-491c294311c401a333e96f63b7fdf3a40f2f16a0d014bd7806d7539e7a99a85cR66, the receiver is not a pointer which means in any scenerio the whole client would never be `nil` as its not a pointer.

Reply:
the state is `unknown` now as that comment was resolved @mik-dass

Reply:
> the state is unknown now as that comment was resolved @mik-dass

I meant the state of the URLs returned from this function as `unknown` since it might not confirm the existence against the cluster.

Reply:
currently we cannot show the urls because as the cluster is not available we cannot decide if the url is a route or ingress hence they cannot be shown

Reply:
opened as issue for `unknown` https://github.com/openshift/odo/issues/4492